azure-pipelines.yml

```diff
@@ -32,24 +28,33 @@ stages:
           Dockerfile: '$(appdir)Dockerfile'
   
   - stage: SonarCloud
+    dependsOn: []
     jobs:
     - job: build
       steps:
+      # using node version 15.x to enforce npm 7.x
       - task: NodeTool@0
         inputs:
-          versionSpec: '10.x'
+          versionSpec: '15.x'
         displayName: 'Install Node.js'
 
-      - script: |
-          npm install -g @angular/cli
-          npm install
-          ng build --prod
-        continueOnError: True
-        displayName: 'npm install and build'
+      - script: npm ci
+        displayName: npm install
+        workingDirectory: $(appDir)
+
+      - script: npx ng build --prod
+        displayName: ng build
+        workingDirectory: $(appDir)
+      
+      - script: npx ng lint
+        displayName: ng lint
+        workingDirectory: $(appDir)
+      
+      - script: npx ng test
+        displayName: ng test
```

Comment:
ng test unadorned won't work, we need --browsers chromeheadless, --no-watch, and we also want code coverage collected & sent to sonarcloud. lcov format is the one it can understand, and --coverage puts it in a coverage folder you can see locally.

the chromeheadless config that the pipeline needs should also work locally so you can iterate on the command until it works locally without waiting on the pipeline to run

Reply:
I apologize, didn't see this review down here. Working on code coverage now